src/vs/workbench/contrib/debug/common/debugModel.ts

```diff
@@ -2003,6 +2014,7 @@ export class DebugModel extends Disposable implements IDebugModel {
 
 	addWatchExpression(name?: string): IExpression {
 		const we = new Expression(name || '');
+		this._register(we.onDidChangeValue((e) => this._onDidChangeWatchExpressionValue.fire(e)));
```

Comment:
This should be disposed when the watch expression is removed

Reply:
https://github.com/microsoft/vscode/pull/222123